app/src/main/java/org/simple/clinic/home/report/ReportsUiRenderer.kt

```diff
@@ -1,12 +1,28 @@
 package org.simple.clinic.home.report
 
 import org.simple.clinic.mobius.ViewRenderer
+import org.simple.clinic.util.ValueChangedCallback
 
 class ReportsUiRenderer(
     private val ui: ReportsUi
 ) : ViewRenderer<ReportsModel> {
 
+  private val valueChangedCallback = ValueChangedCallback<ReportsModel>()
+
   override fun render(model: ReportsModel) {
+    valueChangedCallback.pass(model) {
+      val reportsContent = it.reportsContent
+      if (reportsContent != null) {
```

Comment:
We could extract this to a computed property, say `hasLoadedReports` on `ReportsModel`.